odinmain/src/main/kotlin/me/odinmain/features/impl/floor7/p3/TerminalSounds.kt

```diff
@@ -41,35 +43,66 @@ object TerminalSounds : Module(
     @SubscribeEvent
     fun onPacket(event: PacketReceivedEvent){
         with(event.packet) {
+            //if (this is S29PacketSoundEffect) modMessage("$soundName, $volume, $pitch")
             if (
-                this !is S29PacketSoundEffect ||
-                currentTerm == TerminalTypes.NONE ||
-                customSound == "note.pling" ||
-                soundName != "note.pling" ||
-                volume != 8f ||
-                pitch != 4.047619f ||
-                !clickSounds
-            ) return
-            playTerminalSound()
-            event.isCanceled = true
+                this is S29PacketSoundEffect &&
+                soundName == "note.pling" &&
+                volume == 8f &&
+                //pitch == 1.8888888f &&
+                pitch == 4.047619f &&
+                shouldReplaceSounds
+            ) event.isCanceled = true
         }
     }
 
+    @SubscribeEvent
+    fun onSlotClick(event: GuiEvent.GuiMouseClickEvent) {
+        if (!shouldReplaceSounds) return
+        val slot = (event.gui as? GuiChest)?.slotUnderMouse?.slotIndex ?: return
+        clickSlot(slot)
+    }
+
+    @SubscribeEvent
+    fun onCustomSlotClick(event: GuiEvent.CustomTermGuiClick) {
+        if (!shouldReplaceSounds) return
+        clickSlot(event.slot)
+    }
+
     @SubscribeEvent
     fun onTermComplete(event: TerminalSolvedEvent) {
-        if (event.type == TerminalTypes.NONE || mc.currentScreen is TermSimGui || event.playerName != mc.thePlayer?.name || !completeSounds) return
-        playCompleteSound()
+        if (shouldReplaceSounds && event.playerName != mc.thePlayer?.name || (!completeSounds && !clickSounds)) mc.thePlayer.playSound("note.pling", 8f, 4f)
+        else if (shouldReplaceSounds && completeSounds && !clickSounds) playCompleteSound()
+    }
+
+    init {
+        onMessage("The gate has been destroyed!", false, { enabled && shouldReplaceSounds }) { mc.thePlayer.playSound("note.pling", 8f, 4f) }
+
+        onMessage("The Core entrance is opening!", false, { enabled && shouldReplaceSounds }) { mc.thePlayer.playSound("note.pling", 8f, 4f) }
+    }
+
+    private fun clickSlot(slot: Int) {
+        if (
+            (currentTerm != TerminalTypes.ORDER && slot !in TerminalSolver.solution) ||
+            (currentTerm == TerminalTypes.ORDER && slot != TerminalSolver.solution.first()) ||
+            (currentTerm == TerminalTypes.MELODY && slot !in arrayOf(43, 34, 25, 16))
```

Comment:
uh make the (currentTerm == TerminalTypes.MELODY && slot == 43) check here instead?

Reply:
unless im misunderstanding, the sound should be playing on every melody slot. the slot == 43 is only to check if its the last slot to play the completion sound.

Reply:
aha ok